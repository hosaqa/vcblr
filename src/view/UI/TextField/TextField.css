.TextField {
  display: inline-block;
}

.TextField:disabled,
.TextField[disabled='disabled'] {
  pointer-events: none;
  opacity: 0.6;
}

.TextField__input {
  display: inline-flex;
  align-items: center;
  text-rendering: auto;
  cursor: auto;
  -webkit-appearance: textfield;
  -moz-appearance: textfield;
  -webkit-tap-highlight-color: transparent;
  color: var(--font-primary);
  border: 1px solid var(--border-primary);
  border-radius: var(--border-radius);
  padding: 7px;
  width: 100%;
  max-width: 100%;
}
