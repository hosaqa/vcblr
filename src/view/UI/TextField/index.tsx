import Textfield from './TextField';

export default Textfield;
