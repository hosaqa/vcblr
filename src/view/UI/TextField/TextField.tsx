import React, { memo } from 'react';
import clsx from 'clsx';
import './TextField.css';

interface ITextField extends React.InputHTMLAttributes<HTMLInputElement> {
  className?: string;
}

const TextField: React.FC<ITextField> = memo(({ className, ...props }) => (
  <div className={clsx('TextField', className)}>
    <input className="TextField__input" {...props} />
  </div>
));

export default TextField;
