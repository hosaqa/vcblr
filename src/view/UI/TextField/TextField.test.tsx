import React from 'react';
import { shallow } from 'enzyme';
import TextField from './TextField';

describe('Textfield UI', () => {
  describe('default props', () => {
    const textfield = shallow(<TextField />);

    it('renders properly', () => {
      expect(textfield).toMatchSnapshot();
    });

    it('renders div element', () => {
      expect(textfield.find('.TextField')).toHaveLength(1);
    });

    it('renders input element', () => {
      expect(textfield.find('.TextField .TextField__input')).toHaveLength(1);
    });
  });

  describe('custom class', () => {
    const textfield = shallow(<TextField className="custom" />);

    it('renders properly', () => {
      expect(textfield).toMatchSnapshot();
    });

    it('div element has custom class', () => {
      expect(textfield.find('.TextField')).toHaveLength(1);
      expect(textfield.find('.TextField').hasClass('custom')).toBeTruthy();
    });
  });

  describe('html attributes on input', () => {
    const inputAriaLabel = 'enter text';
    const inputValue = 'default text';

    const textfield = shallow(
      <TextField aria-label={inputAriaLabel} value={inputValue} />
    );

    it('renders properly', () => {
      expect(textfield).toMatchSnapshot();
    });

    it('renders div element', () => {
      expect(textfield.find('.TextField')).toHaveLength(1);
    });

    it('input attributes', () => {
      expect(
        textfield.find('.TextField .TextField__input').prop('aria-label')
      ).toEqual(inputAriaLabel);
      expect(
        textfield.find('.TextField .TextField__input').prop('value')
      ).toEqual(inputValue);
    });
  });
});
