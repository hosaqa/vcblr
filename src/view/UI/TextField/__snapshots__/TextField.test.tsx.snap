// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Textfield UI custom class renders properly 1`] = `ShallowWrapper {}`;

exports[`Textfield UI default props renders properly 1`] = `ShallowWrapper {}`;

exports[`Textfield UI html attributes on input renders properly 1`] = `ShallowWrapper {}`;
